source/mir/math/numeric.d

```diff
@@ -185,7 +186,53 @@ struct ProdAccumulator(T, ProdAlgo prodAlgo)
         ///
         ProdAccumulator!(T, prodAlgo) ldexp(long exp) const
         {
-            return typeof(return)(this.exp + exp, mantissa);
+            return typeof(return)(this._exp + exp, this._mantissa);
+        }
+    }
+
+    static if (prodAlgo == ProdAlgo.separateExponentAccumulation) {
+        ///
+        long exp()
+        {
+            return this._exp;
+        }
+
+    } else static if (prodAlgo == ProdAlgo.naive && 
+                      isFloatingPoint!T) {
```

Comment:
please remove `naive` and simplify the structure. Integers should be accumulated like a double by default.
Later Ion's `Fp` will replace double.